File: server/test_api.py
import urllib.request
import urllib.parse
import json

# Reset admin password via bootstrap-admin
url_bootstrap = "http://localhost:8000/api/bootstrap-admin"
payload_bootstrap = {
    "secret": "dev-admin-secret",
    "correo": "admin@paginas360.com",
    "nombre": "Admin",
    "apellido": "Principal",
    "password": "secret"
}
data_bootstrap = json.dumps(payload_bootstrap).encode("utf-8")
req_bootstrap = urllib.request.Request(url_bootstrap, data=data_bootstrap, method="POST")
req_bootstrap.add_header("Content-Type", "application/json")

try:
    with urllib.request.urlopen(req_bootstrap) as res_boot:
        boot_data = json.loads(res_boot.read().decode("utf-8"))
        print("Bootstrap Response:", boot_data)
        
    # Now, test signin
    url_signin = "http://localhost:8000/api/signin"
    data = urllib.parse.urlencode({"username": "admin@paginas360.com", "password": "secret"}).encode("utf-8")
    req = urllib.request.Request(url_signin, data=data, method="POST")
    req.add_header("Content-Type", "application/x-www-form-urlencoded")

    with urllib.request.urlopen(req) as res:
        res_data = json.loads(res.read().decode("utf-8"))
        print("Signin Response:", res_data)
        token = res_data["access_token"]
        
        # Test mePermisos
        url_perms = "http://localhost:8000/api/me/permisos"
        req_perms = urllib.request.Request(url_perms, method="GET")
        req_perms.add_header("Authorization", f"Bearer {token}")
        
        with urllib.request.urlopen(req_perms) as res_perms:
            perms_data = json.loads(res_perms.read().decode("utf-8"))
            print("me/permisos Response:", perms_data)
except Exception as e:
    print("Error:", e)

File: test_api.js
const http = require('http');

const options = {
  hostname: 'localhost',
  port: 8000,
  path: '/api/signin',
  method: 'POST',
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded'
  }
};

const req = http.request(options, (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const json = JSON.parse(data);
    if (!json.access_token) {
        console.error("Login failed:", json);
        return;
    }
    const token = json.access_token;
    
    // Fetch empresas
    http.get({
      hostname: 'localhost',
      port: 8000,
      path: '/api/empresas/?limit=10',
      headers: { 'Authorization': `Bearer ${token}` }
    }, (res2) => {
      let data2 = '';
      res2.on('data', chunk => data2 += chunk);
      res2.on('end', () => {
        const empresas = JSON.parse(data2);
        console.log("Total empresas:", empresas.length);
        const e10 = empresas.find(e => e.id === 10);
        console.log("Empresa 10:", e10);
      });
    });
  });
});

req.write('username=admin@admin.com&password=12345678');
req.end();
